Sources/PushNotifications.swift

```diff
@@ -89,6 +96,67 @@ import Foundation
     }
     #endif
 
+    /**
+     Set user id.
+
+     - Parameter userId: User id.
+
+     - Precondition: You need to first authenticate the user by providing `beamsTokenProvider` in `start(instanceId:, beamsTokenProvider:)` method. `beamsTokenProvider` should not be nil.
+    */
+    /// - Tag: setUserId
+    @objc public func setUserId(_ userId: String, completion: @escaping (Error?) -> Void) throws {
+        guard let tokenProvider = self.tokenProvider else {
+            throw UserValidationtError.beamsTokenProviderNotSetException
+        }
+
+        let persistenceService: UserPersistable = PersistenceService(service: UserDefaults(suiteName: Constants.UserDefaults.suiteName)!)
+        if let persistedUserId = persistenceService.getUserId(), persistedUserId == userId {
+            return completion(nil)
+        }
+
+        guard persistenceService.getUserId() == nil else {
+            throw UserValidationtError.userAlreadyExists
+        }
+
+        self.preIISOperationQueue.async {
+            guard let deviceId = Device.getDeviceId() else {
+                return completion(TokenProviderError.error("[PushNotifications] - Device id is nil."))
+            }
+            guard let instanceId = Instance.getInstanceId() else {
+                return completion(TokenProviderError.error("[PushNotifications] - Instance id is nil."))
+            }
+            guard let url = URL(string: "https://\(instanceId).pushnotifications.pusher.com/device_api/v1/instances/\(instanceId)/devices/apns/\(deviceId)/user") else {
+                return completion(TokenProviderError.error("[PushNotifications] - Error while constructing URL from a string."))
+            }
+
+            tokenProvider.fetchToken(userId: userId, completionHandler: { (token, error) in
+                guard error == nil else {
+                    return completion(error)
+                }
+
+                let networkService: PushNotificationsNetworkable = NetworkService(session: self.session)
+                networkService.setUserId(url: url, token: token, completion: { _ in
+                    persistenceService.setUserId(userId: userId)
+                    completion(nil)
+                })
+            })
+        }
+    }
+
+    /**
+     Log out user and remove all subscriptions.
+
+     Remove user and all subscriptions set from both client and server.
+     Device is now in a fresh state, ready for new subscriptions or user being set.
+     */
+    /// - Tag: clearAllState
+    @objc public func clearAllState() {
+        // TODO: Missing API call!
```

Comment:
FYI, this is now available, @lukabratos.

Reply:
Implemented in 3268c0a2848894f6cab19338e57dfa5a3ee2c050